internal/infrastructure/kubernetes/infra_client.go

```diff
@@ -27,17 +28,16 @@ func (cli *InfraClient) CreateOrUpdate(ctx context.Context, key client.ObjectKey
 		if kerrors.IsNotFound(err) {
 			// Create if it does not exist.
 			if err := cli.Client.Create(ctx, specific); err != nil {
-				return err
+				return fmt.Errorf("for Create %v", err)
```

Comment:
that's the different with before, how can this fix the issue?

Reply:
![image](https://github.com/envoyproxy/gateway/assets/2174082/22656775-04a7-4d94-9e8e-bb9e023b14cb)

Reply:
can you explain more about the detail about this edge case?

Reply:
xref: https://stackoverflow.com/questions/65987577/kubectl-apply-reports-error-operation-cannot-be-fulfilled-on-serviceaccounts

we need to update existing resources so keep `uuid`.  the `resourceVersion` will update at every editing.

Reply:
xref: #1041

Reply:
should we use retry.OnConflict here?

Reply:
did it

Reply:
I  think retryOnConflict is unnecessary. the `resourceVersion` has been removed.